cmd/mdatagen/internal/sampleprocessor/doc.go

```diff
@@ -3,6 +3,7 @@
 
 // Generate a test metrics builder from a sample metrics set covering all configuration options.
 //go:generate mdatagen metadata.yaml
+//go:generate mdatagen metadata_custom.yaml
```

Comment:
Let's avoid the duplication. Maybe having a custom package in the processor and leaving the receiver as is.

Reply:
I opted to remove duplication slightly differently in [f106762](https://github.com/open-telemetry/opentelemetry-collector/pull/11468/commits/f106762ff9919469a61dd49341e6c04c8957d63b).

I left the custom package in `samplereceiver`. This still ensures the extent of coverage:

* Generates the representative set of templates for a metadata package
* Demonstrates the capability of generating two separate packages with different names

The custom package in `sampleprocessor` was not adding anything to the above coverage by just doing it in `samplereceiver`, so I removed that. I think the current duplication is acceptable since it still provides the necessary coverage, but if you disagree I can change it.

Reply:
I still don't understand why we need two packages for the same receiver. This will be confusing to maintain going forward. Also, we need to show how mdatagen is supposed to be used. Having two yaml files and two packages for the same module is not something we should recommend. Can we just add `generated_package_name` to the existing yaml for the receiver and move the package to a custom path?

Reply:
Having two yaml files and two packages is the reason the custom package name feature was added in the first place. It was the only reasonable way I could see for a receiver to have a featuregated semantic convention transition, as doing it within one metadata package seems infeasible based on my attempts. We wouldn't want two packages to exist forever, but for the lifetime of the featuregate (which for a semconv transition will likely be a longer horizon) two packages would need to coexist.